pkg/kubelet/pod_workers.go

```diff
@@ -99,6 +99,10 @@ const (
 	// TerminatedPodWork indicates the pod is stopped, can have no more running
 	// containers, and any foreground cleanup can be executed.
 	TerminatedPodWork
+	// TemporarilyTerminatedPodWork is the same as TerminatedPodWork, but indicates
+	// that a create update was delivered AFTER the pod was terminated, indicating
+	// that the pod may have been recreated with the same UID.
+	TemporarilyTerminatedPodWork
```

Comment:
Comment above on L89 needs to be updated... I find this name a little confusing, given the context in which this happens perhaps "RecreatePodWork" or "RestartedPodWork" would be better?

Reply:
There is no guarantee of restart though, and technically to the kubelet is a different pod (so it's not a restart).  RecreatePodWork feels like it is implying that the pod worker wants to restart the pod.  `TerminatedWithRecreatePodWork` might be clearer.  I want to preserve that the pod is definitely stopped at the moment and will not restart without intervention.  This enum is a bit tricky though - I might create another enum that is similar but clearer.

Reply:
In the latest commit.